.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,10 @@
+name: CI Github Actions
+on: [push]
+jobs:
+ build:
+  name: Build and test
+  runs-on: ubuntu-latest
+  steps:
+  - uses: actions/checkout@v2
+  - run: docker build --target test --tag todo-app:test . 
+  - run: docker run todo-app:test
```

Comment:
This looks good but your tests are [currently failing in your pipeline](https://github.com/cyberspark13/DevOps-Course-Starter/actions/runs/3098312165/jobs/5016066856)!

The reason is that your end-to-end tests require some real secrets to run, but they don't have them here. To fix, I'd recommend first splitting your command to run unit/integration tests & end-to-end tests into two parts, which we can do by telling pytest which folders to run the tests in:
```
  - run: docker run todo-app:test tests
  - run: docker run todo-app:test tests_e2e
```
That should get the "tests" folder passing but the e2e tests will still have the same issue. We can then look at passing in the relevant secrets/environment variables; you'll need to set the values in the repository secrets, and then pass them through like below (but with all of the relevant variables included)
```
- run: docker run -e TRELLO_API_KEY=${{ secrets.TRELLO_API_KEY }} -e TRELLO_API_SECRET.... todo-app:test tests_e2e
```